nameko_grpc/entrypoint.py

```diff
@@ -165,7 +165,18 @@ def handle_request(self, request_stream, response_stream):
         request = request_stream.consume(self.input_type)
 
         if self.cardinality in (Cardinality.UNARY_STREAM, Cardinality.UNARY_UNARY):
-            request = next(request)
+            try:
+                request = next(request)
+            except Exception:
+                exc_info = sys.exc_info()
+                message = "Exception deserializing request!"
+                error = GrpcError.from_exception(
+                    exc_info,
+                    code=StatusCode.INTERNAL,
+                    message=message,
+                )
+                response_stream.close(error)
+                return
```

Comment:
Might be easier and a bit more DRY to catch this exception and then send it to `handle_result`?

I also don't think it's accurate to always describe the error that could be happening here as a problem deserialising. "Exception calling application" as in `handle_result` is generic but probably more correct.

Reply:
We seem to be promising to return the same status error codes as the `grpc` library - this is enforced by the tests including the one I added.

I initially had something more like what you're describing, but we don't really have the correct inputs for `handle_result` available at this point, as well as the error code issue.

Reply:
Ha, I just noticed that the `grpc` library uses that message. Cool as it is then.